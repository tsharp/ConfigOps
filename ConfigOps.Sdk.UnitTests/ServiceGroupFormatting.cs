using ConfigOps.Core.Serialization;
using ConfigOps.Sdk.Api.v1alpha1;
using Snapshooter.Xunit;

namespace ConfigOps.Sdk.UnitTests
{
    public class ServiceGroupFormatting
    {
        [Fact]
        public void Validate_AwsServiceGroup_YamlFormat()
        {
            ServiceGroup serviceGroup = new ServiceGroup()
            {
                ApiVersion = "config.ops.kuipersys.com/v1alpha1",
                Metadata =
                {
                    Name = "service-example-uswest1",
                    Namespace = "amazon-aws"

                },
                Spec =
                {
                    CloudProvider = "amazon-aws/cloud-config",
                    CloudProviderRegion = "amazon-aws/us-west-1",
                    Settings =
                    {
                        { "secretStoreXyz", "abcd" }
                    }
                }
            };

            serviceGroup
                .SerializeToYaml()
                .MatchSnapshot();
        }
    }
}
