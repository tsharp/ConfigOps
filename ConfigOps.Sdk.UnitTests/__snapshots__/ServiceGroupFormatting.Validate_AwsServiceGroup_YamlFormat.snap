apiVersion: config.ops.kuipersys.com/v1alpha1
kind: ServiceGroup
metadata:
  name: service-example-uswest1
  namespace: amazon-aws
spec:
  cloudProvider: amazon-aws/cloud-config
  cloudProviderRegion: amazon-aws/us-west-1
  settings:
    secretStoreXyz: abcd
